### ISSUE TITLE
Can we Do a transfert learning with a new label

### ISSUE THREAD

**Issue by kamelbouyacoub** [thtrieu/darkflow#117]
Hello 

I had trained the model with 2 classes, and I want to use the generated .ckpt, to train the model woth news classes, How can I do that, because when I test it with a new configuration, the prgrmm didn't work 

Thanks

**Comment by Prakash19921206** [comment]
this would be a great idea if transfer learning is possible with yolo.weights...
@thtrieu, is this possible?
 i couldn't find much info in [here](https://thtrieu.github.io/notes/yolo-tensorflow-graph-buffer-cpp)

**Comment by Surakshaajith** [comment]
Yup, training time would be reduced if transfer learning is introduced. This will be helpful for many.

**Comment by kamelbouyacoub** [comment]
Is there anyone who know how to do a transfer learning in tensorflow ?

**Comment by jcarletgo** [comment]
I think you could you modify the code in utils/loader.py to be able to load a specified .ckpt file. Perhaps use '--load ./ckpt/trained-net-###.meta'

**Comment by thtrieu** [comment]
Originally the code was able to load ckpt (and thus do transfer learning), then tensorflow 1.0 breaks backward compatibility and until now I have not the time to fix the old code.

In fact what @jcarletgo  said about modifying the loader is correct.

**Comment by abagshaw** [comment]
@thtrieu Just wondering if this functionality is still missing from darkflow? I was under the impression that one could load from a ckpt file with darkflow to either `--savepb` or continue training.

On a similar note, let's imaging I have a bunch of images with faces annotated. If I wanted to say add a "face" class to the already trained `yolo.weights` (add a face class onto the existing 80 COCO classes) I would have to create a new `.cfg` file (say `yolo-new.cfg`) based off of `yolo.cfg` with some modifications to the last two layers to accommodate the new class if I understand this correctly. The weights that correspond to the identical layers between the two `.cfg` files will be kept (but the weights for the last two will be tossed out?). If that is so then I would suppose that I would have to retrain with the whole COCO dataset + my faces dataset to get the detection ability I am looking for - or am I misunderstanding something?

Also if my faces dataset is only annotated for faces - will that cause issues with predictions for the `person` dataset as the people in those images are not annotated with a `person` box which may incorrectly teach the net that they are not people?

**Comment by infinyte** [comment]
I am trying to evaluate the possibility of developing a large scale solution which requires some combination of transfer learning and real time object recognition.  I have been following the current research and it seems like enhancing/extending the YOLO weighting stuff could be a good solution.  If I wanted to contribute my Tensorflow exploratory efforts, provided they are working reasonably well of course, who is the point of contact here?

**Comment by abagshaw** [comment]
@thtrieu Is the point of contact, although I believe he is pretty busy right now. If you have something working, feel free to open a pull request though!
